app/src/main/java/com/auth0/googleplus/testapp/MainActivity.java

```diff
@@ -0,0 +1,148 @@
+
+import com.auth0.api.authentication.AuthenticationAPIClient;
+import com.auth0.api.authentication.AuthenticationRequest;
+import com.auth0.api.callback.AuthenticationCallback;
+import com.auth0.core.Auth0;
+import com.auth0.core.Strategies;
+import com.auth0.core.Token;
+import com.auth0.core.UserProfile;
+import com.auth0.googleplus.GooglePlusIdentityProvider;
+import com.auth0.identity.IdentityProviderCallback;
+
+public class MainActivity extends AppCompatActivity {
+
+    private static final String TAG = MainActivity.class.getName();
+
+    GooglePlusIdentityProvider provider;
+
+    boolean authRequestInProgress;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        final TextView textView = (TextView) findViewById(R.id.textView);
+
+        Auth0 auth0 = new Auth0(getString(R.string.auth0_client_id), getString(R.string.auth0_domain_name));
+        final AuthenticationAPIClient client = new AuthenticationAPIClient(auth0);
+
+        provider = new GooglePlusIdentityProvider(MainActivity.this);
+        provider.setCallback(new IdentityProviderCallback() {
+            @Override
+            public void onFailure(Dialog dialog) {
+                Log.d(TAG, "onFailure, showing dialog...");
+
+                textView.setText("");
+                dialog.show();
+            }
+
+            @Override
+            public void onFailure(int titleResource, int messageResource, Throwable cause) {
+                Log.d(TAG, "onFailure, titleResource: " + titleResource + ", messageResource: " + messageResource + ", cause: " + cause);
+
+                textView.setText("");
+                AlertDialog dialog = new AlertDialog.Builder(MainActivity.this)
+                        .setTitle(titleResource)
+                        .setMessage(messageResource)
+                        .create();
+                dialog.show();
+            }
+
+            @Override
+            public void onSuccess(String serviceName, String accessToken) {
+                Log.d(TAG, "onSuccess, serviceName: " + serviceName + ", accessToken: " + accessToken);
+
+                textView.setText("Trying to log in with GooglePlus token: " + accessToken);
+
+                AuthenticationRequest request = client.loginWithOAuthAccessToken(accessToken, serviceName);
+                request.start(new AuthenticationCallback() {
+                    @Override
+                    public void onSuccess(UserProfile profile, Token token) {
+                        textView.setText("Welcome " + profile.getName());
+                    }
+
+                    @Override
+                    public void onFailure(Throwable error) {
+                        textView.setText("Log in failed. " + error.getMessage());
+                    }
+                });
+            }
+
+            @Override
+            public void onSuccess(Token token) {
```

Comment:
yep, only works for webview based auth cause we get the auth0 token from the url